code/modules/recycling/disposal-unit.dm

```diff
@@ -527,6 +527,160 @@
+	..()
+	stored.ptype = DISP_END_CHUTE
+	spawn(5)
+		trunk = locate() in loc
+		if(trunk)
+			trunk.linked = src
+
+/obj/machinery/disposal/trapdoor/Crossed(AM as mob|obj)
+	if(open)
+		if(istype(AM, /mob/living/))
+			var/mob/living/M = AM
+			if(M.floating)
+				return
+			M.loc = src
+			trap_flush()
+			if(auto_close_on_mob)
+				spawn(auto_close_on_mob)
+					close()
+			return
+		if(istype(AM, /obj/item))
+			var/obj/item/O = AM
+			spawn(5)
+				if(O.throwing || O.anchored)
+					return
+				else if(O.loc == src.loc)
+					O.loc = src
+					trap_flush()
+
+/obj/machinery/disposal/trapdoor/MouseDrop_T(mob/living/target, mob/living/user)
+	if (!open)
+		return
+	if(istype(target) && user == target)
+		push_mob_in(target, user)
+		return 1
+
+/obj/machinery/disposal/trapdoor/proc/push_mob_in(mob/living/target, mob/living/user)
+	if(target.buckled)
+		return
+	add_fingerprint(user)
+	if(user == target)
+		if(target.floating)
+			user.visible_message("[user] is attempting to dive into [src].", \
+				"<span class='notice'>You start diving into [src]...</span>")
+			if(do_mob(target, user, 10))
+				if (!loc)
+					return
+				target.loc = src
+				user.visible_message("[user] dives into [src].", \
+					"<span class='notice'>You dive into [src].</span>")
+				update()
+				sleep(5)
+				trap_flush()
+				if(auto_close_on_mob)
+					spawn(auto_close_on_mob)
+						close()
+		else
+			user.visible_message("[user] is attempting to step on the edge of [src].", \
+				"<span class='notice'>You start attempting to step on the edge of [src]...</span>")
+			if(do_mob(target, user, 30))		//Add 25% chance of failing "You slip and fall in to [src], 75% if clumsy "You step.. nope too funny you backwards flip in to [src], HONK!", 75% if braindamaged "You close your eyes so [src] cant detect you, and boldly step forward.", 100% if wet floor (noslip/galosh check) "You slip, and fall in to [src]."
+				if (!loc)	//Chance of getting damaged on failure? "You slip and land face first on the edge, knocking X teeths off."
+					return
+				target.loc = src.loc
+				user.visible_message("[user] steps on the edge of [src].", \
+					"<span class='notice'>You step on the edge of [src].</span>")
+				update()	//Does not seem to work even at parent, maybe use grab and showe in instead?
+	if(user != target)
+		target.visible_message("<span class='danger'>[user] starts pushing [target] into [src].</span>", \
+			"<span class='userdanger'>[user] starts pushing you into [src]!</span>")
+		user.visible_message("<span class='notice'>You start pushing [target] into [src]...</span>")
+		if(do_mob(target, user, 10))
+			if (!loc)
+				return
+			target.loc = src
+			target.visible_message("<span class='danger'>[user] has pushed [target] in [src].</span>", \
+				"<span class='userdanger'>[user] has pushedd [target] in [src].</span>")
+			add_logs(user, target, "pushed", addition="into [src]")
```

Comment:
this doesn't cause any harm does it, not sure if logging is needed but hey,i guess